src/main/java/com/segment/analytics/android/integrations/nielsendcr/NielsenDCRIntegration.java

```diff
@@ -0,0 +1,456 @@
+
+      case "Video Content Completed":
+        stopPlayheadTimer();
+        break;
+    }
+  }
+
+  public void trackVideoAd(
+      TrackPayload track, Properties properties, Map<String, Object> nielsenOptions)
+      throws JSONException {
+    String event = track.event();
+
+    Map<String, String> adMapper = new LinkedHashMap<>();
+    adMapper.put("assetId", "assetid");
+    adMapper.put("type", "type");
+    adMapper.put("title", "title");
+
+    switch (event) {
+      case "Video Ad Started":
+        // In case of ad `type` preroll, call `loadMetadata` with metadata values for content,
+        // followed by `loadMetadata` with ad (preroll) metadata
+        if ("pre-roll".equals(properties.getString("type"))) {
+          JSONObject adContentAsset = buildAdContentMetadata(properties, nielsenOptions, adMapper);
+          appSdk.loadMetadata(adContentAsset);
+          logger.verbose("appSdk.loadMetadata(%s)", adContentAsset);
+        }
+        JSONObject adAsset = buildAdMetadata(properties, adMapper);
+        appSdk.loadMetadata(adAsset);
+        logger.verbose("appSdk.loadMetadata(%s)", adAsset);
+        startPlayheadTimer(properties, appSdk);
+        break;
+
+      case "Video Ad Playing":
+        startPlayheadTimer(properties, appSdk);
+        break;
+
+      case "Video Ad Completed":
+        stopPlayheadTimer();
+        appSdk.stop();
+        logger.verbose("appSdk.stop");
+        break;
+    }
+  }
+
+  @Override
+  public void track(TrackPayload track) {
+    String event = track.event();
+    Properties properties = track.properties();
+
+    Map<String, Object> nielsenOptions = track.integrations().getValueMap("nielsen-dcr");
+    if (isNullOrEmpty(nielsenOptions)) {
+      nielsenOptions = Collections.emptyMap();
+    }
+
+    switch (event) {
+      case "Video Playback Started":
+      case "Video Playback Paused":
+      case "Video Playback Interrupted":
+      case "Video Playback Buffer Started":
+      case "Video Playback Buffer Completed":
+      case "Video Playback Resumed":
+        try {
+          trackVideoPlayback(track, properties, nielsenOptions);
+        } catch (JSONException e) {
+          logger.verbose("Error tracking Video Playback:", e);
+        }
+        break;
+      case "Video Content Started":
+      case "Video Content Playing":
+      case "Video Content Completed":
+        try {
+          trackVideoContent(track, properties, nielsenOptions);
+        } catch (JSONException e) {
+          logger.verbose("Error tracking Video Content:", e);
+        }
+        break;
+      case "Video Ad Started":
+      case "Video Ad Playing":
+      case "Video Ad Completed":
+        try {
+          trackVideoAd(track, properties, nielsenOptions);
+        } catch (JSONException e) {
+          logger.verbose("Error tracking Video Ad:", e);
+        }
+        break;
+    }
+  }
+
+  @Override
+  public void screen(ScreenPayload screen) {
+    String name = screen.name();
```

Comment:
What if no screen name is set?

Reply:
I don't think this will be an issue. We fallback to `category` if `name` is not set, and [throw an exception](https://github.com/segmentio/analytics-android/blob/a6d77636bd19981e749646bf22f26b8f2646ee5c/analytics/src/main/java/com/segment/analytics/integrations/ScreenPayload.java#L153) if neither are set